lib/trino-plugin-toolkit/src/main/java/io/trino/plugin/base/security/FileBasedAccessControl.java

```diff
@@ -405,6 +406,14 @@ public void checkCanCreateMaterializedView(ConnectorSecurityContext context, Sch
         }
     }
 
+    @Override
+    public void checkCanRefreshMaterializedView(ConnectorSecurityContext context, SchemaTableName materializedViewName)
+    {
+        if (!checkTablePermission(context, materializedViewName, INSERT)) {
```

Comment:
Do we have a documentation page where we could describe that MV refresh requires `INSERT` permission?

Reply:
There is no documentation around MVs in general

Reply:
I think it should be `OWNERSHIP`

Reply:
There could be a need for a setup where only a restricted group of admins can create and drop MVs whereas a larger group of end users are able to use the MVs or REFRESH them but not create and drop them.
I think we wouldn't have that flexibility if require OWNERSHIP permission.

Reply:
Discussed offline. Let's use UPDATE.

Reply:
@raunaqmorarka You are probably already aware. Here is the spec that can hopefully help guide docs: https://docs.google.com/document/d/10jPGw3t-Tu8OgWo5oC9d-O8d1PVdnAbEhyfvnJA0T8U/edit

Reply:
We used `UPDATE` because refresh can both drop and insert data, therefore `INSERT` permissions itself is not accurate.